publication-commons/src/main/java/no/unit/nva/publication/commons/customer/JavaHttpClientCustomerApiClient.java

```diff
@@ -0,0 +1,60 @@
+package no.unit.nva.publication.commons.customer;
+
+import static java.net.HttpURLConnection.HTTP_OK;
+import static nva.commons.core.attempt.Try.attempt;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.Builder;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandlers;
+import no.unit.nva.auth.AuthorizedBackendClient;
+import no.unit.nva.auth.CognitoCredentials;
+import no.unit.nva.commons.json.JsonUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JavaHttpClientCustomerApiClient implements CustomerApiClient {
+    private static final Logger LOGGER = LoggerFactory.getLogger(JavaHttpClientCustomerApiClient.class);
+
+    private final HttpClient httpClient;
+    private final CognitoCredentials cognitoCredentials;
+
+    public JavaHttpClientCustomerApiClient(final HttpClient httpClient,
+                                           final CognitoCredentials cognitoCredentials) {
+        this.httpClient = httpClient;
+        this.cognitoCredentials = cognitoCredentials;
+    }
+
+    @Override
+    public Customer fetch(URI customerId) {
```

Comment:
Hvem sin customer spør vi om? Brukerens (som kan være redaktør på en annen institusjon) eller resourceOwners?

Reply:
I CreatePublicationHandler spør vi med `currentCustomerId` fra token/userInfo. Ved UpdatePublicationHandler bruker `p.publisher` med den antagelsen at den kunden som publiserte posten opprinnelig er den som posten skal følge config for. Med config menes da publiseringsstrategi/workflow of filesAllowedForTypes.

Denne endringen gjør at vi aldri kan slette en kunde etter at kunden har publisert noe. Da vil man ikke kunne redigere posten etterpå. Så kunder må deaktiveres. Jeg kan initiere en diskusjon på dette på slack.